nixos/modules/services/home-automation/home-assistant.nix

```diff
@@ -388,9 +446,24 @@ in {
           rm -f "${cfg.configDir}/ui-lovelace.yaml"
           ln -s ${lovelaceConfigFile} "${cfg.configDir}/ui-lovelace.yaml"
         '';
+        copyCustomLovelaceModules = if cfg.customLoveLaceModules != [] then ''
+          mkdir -p "${cfg.configDir}/www"
+          ln -sf ${customLovelaceModulesDir} "${cfg.configDir}/www/nixos-lovelace-modules"
```

Comment:
IIRC if `${cfg.configDir}/www/nixos-lovelace-modules` already exists, then this will create a new symlink underneath it.  Perhaps best to do the `rm -f` in all cases and then only recreate the dir if `customLovelaceModules` is non-empty?

Reply:
I guess none of this would be a problem if the whole `cfg.configDir` was fully declarative, but we can't do that since HA keeps state in there...

Reply:
Good catch. We could add `-n` instead.

>   -n, --no-dereference        treat LINK_NAME as a normal file if
>                                it is a symbolic link to a directory

Reply:
Yes that works!